### PR TITLE
Journal

### PR DESCRIPTION
**Pull request by supernova-97**
Basic construct of the journal, including an overview with dynamic routing

### CODE REVIEW

File: components/JournalForm/JournalForm.js
Comment (PReichetanz) [inline components/JournalForm/JournalForm.js:1]: Following the naming convention you've used before, this file should be named `components/JournalForm/index.js` (and this is better for imports as well).

File: components/JournalForm/JournalForm.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import styled from "styled-components";
+
+function JournalForm({ onJournalFormInput, onJournalSubmit }) {
+  return (
+    <>
+      <StyledForm onSubmit={onJournalSubmit}>
```
Comment (PReichetanz) [inline components/JournalForm/JournalForm.js]: You normally want to call `event.preventDefault` in this component where the submit happens, something like this:

```suggestion
  function handleSubmit(event) {
    event.preventDefault();
    onJournalSubmit();
  }

  return (
    <>
      <StyledForm onSubmit={handleSubmit}>
```

File: pages/journal/form.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  function handleJournalSubmit(e) {
+    e.preventDefault();
+
+    if (
+      journalData.name === "" ||
+      journalData.destination === "" ||
+      journalData.description === ""
+    ) {
```
Comment (PReichetanz) [inline pages/journal/form.js]: Have you tried to use the `required` keyword on the `input` elements instead?

File: pages/journal/form.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      return alert("Missing info!");
+    }
+    setJournals([...journals, { id: uid(), ...journalData }]);
+    router.push("/journal");
```
Comment (PReichetanz) [inline pages/journal/form.js]: Do you need to reset `journalData` here as well?

File: pages/journal/index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import Link from "next/link";
+import styled from "styled-components";
+
+function Journal({ journalData, journals }) {
```
Comment (PReichetanz) [inline pages/journal/index.js]: In order to distinguish this page from a component:
```suggestion
function JournalPage({ journalData, journals }) {
```

File: pages/journal/index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+              ))}
+            </>
+          ) : (
+            "No Journals yet. Create one!"
+          )}
+          <NewJournalButton href="/journal/form">New Journal</NewJournalButton>
+        </EntriesWrapper>
+        <DashboardLink href="/">Dashboard</DashboardLink>
+      </JournalWrapper>
+    </>
```
Comment (PReichetanz) [inline pages/journal/index.js]: Just deleted the fragment:
```suggestion
      <JournalWrapper>
        <h1>Journal</h1>
        <EntriesWrapper>
          {journals.length > 0 ? (
            <>
              {journals.map((journal) => (
                <JournalLink key={journal.id} href={`/journal/${journal.id}`}>
                  {journal.name} in {journal.destination}
                </JournalLink>
              ))}
            </>
          ) : (
            "No Journals yet. Create one!"
          )}
          <NewJournalButton href="/journal/form">New Journal</NewJournalButton>
        </EntriesWrapper>
        <DashboardLink href="/">Dashboard</DashboardLink>
      </JournalWrapper>
```

File: pages/journal/index.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+export default Journal;
+
+const JournalWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const EntriesWrapper = styled.div`
```
Comment (PReichetanz) [inline pages/journal/index.js]: Is a `section` element useful somewhere here?

**Review by PReichetanz — COMMENTED** [review]
Nice work! Only some issues to discuss, but nothing serious.

File: pages/journal/form.js (hunk 3 above)
Comment (supernova-97) [reply pages/journal/form.js]: @PReichetanz I'm not sure what you mean by that

Comment (ella-blaupause) [reply pages/journal/form.js]: @supernova-97 I also did not understand this part

File: pages/journal/form.js (hunk 2 above)
Comment (ella-blaupause) [reply pages/journal/form.js]: Let me know when you have edited the part

**Pull request merged**